x-pack/plugins/canvas/canvas_plugin_src/functions/common/joinRows.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ExpressionFunction } from '../../../../../../src/legacy/core_plugins/interpreter/types';
+import { Datatable } from '../types';
+import { getFunctionErrors } from '../../strings/functions';
+
+interface Arguments {
+  column: string;
+  separator: string;
+  quote: string;
+}
+
+const escapeString = (data: string, quotechar: string): string => {
+  if (quotechar === undefined || quotechar === '') {
+    return data;
+  } else {
+    const regex = new RegExp(quotechar, 'g');
+    return data.replace(/\\/g, `\\\\`).replace(regex, `\\${quotechar}`);
+  }
+};
+
+export function joinRows(): ExpressionFunction<'joinRows', Datatable, Arguments, string> {
+  const errors = getFunctionErrors().getCell;
+  return {
+    name: 'joinRows',
+    type: 'string',
+    help: 'Join values from rows in a datatable into a string',
+    context: {
+      types: ['datatable'],
+    },
+    args: {
+      column: {
+        aliases: ['_'],
+        types: ['string'],
+        required: true,
+        help: 'Column to join values from',
+      },
+      separator: {
+        aliases: ['sep'],
+        types: ['string'],
+        default: ',',
+        help: 'Separator to use between row values',
+      },
+      quote: {
+        types: ['string'],
+        default: 'DEFAULT',
```

Comment:
You have to wrap `"'"` in back ticks to make it work as a default value. From my understanding of the interpreter when the default value is a string, like `default: "'"`, the interpreter tries to parse the content between the quotes as an expression instead of a string, i.e. it'll try to parse something like `quote='` which is invalid syntax. By wrapping `"'"` in backticks, the interpreter will parse `quote="'"` which is what we want here.
```suggestion
        default: `","`,
```

Reply:
Oh, that seems to make it join with the literal value so it becomes
`"'"product1"'","'"product2"'"` instead of `'product1','product2'`

Reply:
In unit tests 
```
default: `\'`
```
works fine but not when running in the browser because the default value gets passed through the interpreter. Does it need to go through the interpreter?